PDFCLOUD-5202 Add Ruby examples for pdfRest API integration

datalogics-kam opened the pull request:
- Add JSON and Multipart payload examples demonstrating API usage.
- Include `.env.example` and integrate `dotenv` for API key management.
- Add `.gitignore` tailored for Ruby projects.
- Introduce `Gemfile` and `Gemfile.lock` for dependency management.
- Provide comprehensive documentation in `README.md`.

datalogics-kam committed:
Add Ruby examples for pdfRest API integration

- Add JSON and Multipart payload examples demonstrating API usage.
- Include `.env.example` and integrate `dotenv` for API key management.
- Add `.gitignore` tailored for Ruby projects.
- Introduce `Gemfile` and `Gemfile.lock` for dependency management.
- Provide comprehensive documentation in `README.md`.

datalogics-kam committed:
Add AGENTS.md outlining repository guidelines and best practices

- Document project structure, coding conventions, and testing process.
- Provide instructions for setup, development, and sample execution.
- Include commit, PR guidelines, and security tips.

datalogics-kam committed:
Add support for EU/GDPR API endpoint configuration

- Update `README.md` with details on optional EU API endpoint for GDPR compliance.
- Extend `.env.example` to include configurable `PDFREST_URL` for overriding the default API base URL.
- Refactor Ruby examples to dynamically use the base URL from environment variables.

datalogics-kam committed:
Update AGENTS.md to include base URL configuration guidelines

- Specify use of `ENV['PDFREST_URL']` for setting the API base URL (defaulting to `https://api.pdfrest.com`).

datalogics-kam committed:
Add Ruby script for merging multiple file types into a single PDF

- Introduce `merge-different-file-types.rb` demonstrating file conversion and merging via pdfRest API.
- Leverage `.env` for API key and base URL configuration.
- Support various file types with content type detection and dynamic conversion to PDF.
- Updated README.md

datalogics-tsmith commented on Ruby/README.md:
```diff
@@ -0,0 +1,193 @@
+
+---
+
+## Roadmap
```
I think the plan is to keep these small so we probably shouldn't have this section so it doesn't look like we intended to do more and then abandoned it

datalogics-tsmith commented on Ruby/README.md:
```diff
@@ -0,0 +1,193 @@
+- Complex Flow Examples/
+    - Higher-level workflows that chain multiple endpoints, e.g., converting mixed input types and merging results.
+
+As more samples are added (expect dozens), you can:
```
I do not think we currently intend to add more

datalogics-kam committed:
Add standardized headers and EU endpoint setup to Ruby examples

- Standardize header comments across Ruby samples for clarity and consistency.
- Update `.env.example` with API key instructions and optional EU endpoint configuration.
- Enhance Ruby examples with detailed setup, usage, and output instructions.
- Revise `AGENTS.md` with audience and tone guidelines for sample development.

datalogics-tsmith commented on Ruby/.env.example:
```diff
@@ -0,0 +1,11 @@
+
+## Optional: Override API base URL/region
+# If unset, scripts default to the US endpoint: https://api.pdfrest.com
+# To route calls to the EU region (for GDPR compliance and lower latency in Europe), set:
```
Can we add a comment directing them to the EU/GDPR section on the site here?

`For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work`

datalogics-kam replied:
Let me know if the positioning is ok, if not, I'll ask the AI to adjust it again.

datalogics-tsmith replied:
I think that's good

datalogics-tsmith commented on Ruby/Complex Flow Examples/merge-different-file-types.rb:
```diff
@@ -0,0 +1,122 @@
+# Setup (.env):
+# - Copy .env.example to .env
+# - Set PDFREST_API_KEY=your_api_key_here
+# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
```
Can we add a comment directing them to the EU/GDPR section on the site here?

`For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work`

datalogics-tsmith commented on Ruby/Endpoint Examples/JSON Payload/markdown.rb:
```diff
@@ -0,0 +1,84 @@
+# Setup (.env):
+# - Copy .env.example to .env
+# - Set PDFREST_API_KEY=your_api_key_here
+# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
```
Can we add a comment directing them to the EU/GDPR section on the site here?

`For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work`

datalogics-tsmith commented on Ruby/Endpoint Examples/JSON Payload/rasterized-pdf.rb:
```diff
@@ -0,0 +1,86 @@
+# Setup (.env):
+# - Copy .env.example to .env
+# - Set PDFREST_API_KEY=your_api_key_here
+# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
```
Can we add a comment directing them to the EU/GDPR section on the site here?

`For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work`

datalogics-tsmith commented on Ruby/Endpoint Examples/Multipart Payload/markdown.rb:
```diff
@@ -0,0 +1,66 @@
+# Setup (.env):
+# - Copy .env.example to .env
+# - Set PDFREST_API_KEY=your_api_key_here
+# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
```
Can we add a comment directing them to the EU/GDPR section on the site here?

`For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work`

datalogics-tsmith commented on Ruby/Endpoint Examples/Multipart Payload/rasterized-pdf.rb:
```diff
@@ -0,0 +1,66 @@
+# Setup (.env):
+# - Copy .env.example to .env
+# - Set PDFREST_API_KEY=your_api_key_here
+# - Optional: set PDFREST_URL to override the API region. For EU/GDPR compliance and proximity, use:
```
Can we add a comment directing them to the EU/GDPR section on the site here?

`For more information visit https://pdfrest.com/pricing#how-do-eu-gdpr-api-calls-work`

datalogics-kam committed:
Add GDPR API call information links to examples and `.env.example`

- Included URL for detailed GDPR compliance info across Ruby examples, `AGENTS.md`, and `.env.example`.
- Enhanced comments to guide users on EU region API usage.
